packages/checkout-extension/src/ExtensionService.ts

```diff
@@ -129,6 +131,21 @@ export class ExtensionService {
                 );
             }
         });
+
+        Object.values(queryHandlerFactories).forEach((createQueryHandlerFactory) => {
+            const handlerFactory = createQueryHandlerFactory(handlerProps);
+
+            if (this.isQueryHandler(handlerFactory.queryType, handlerFactory)) {
+                this.handlers[extension.id].push(
+                    this.checkoutService.handleExtensionQuery(
+                        extension.id,
+                        handlerFactory.queryType,
+                        // eslint-disable-next-line @typescript-eslint/no-misused-promises
```

Comment:
Here should be updated as returning `Promise<void>`.
https://github.com/bigcommerce/checkout-sdk-js/blob/9c4d9b738484cc4eca9506a3dbac720b6ca75e22/packages/core/src/checkout/checkout-service.ts#L1488